codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamChannel.java

```diff
@@ -838,6 +842,9 @@ private void updateLocalWindowIfNeeded() {
                 int bytes = flowControlledBytes;
                 flowControlledBytes = 0;
                 ChannelFuture future = write0(parentContext(), new DefaultHttp2WindowUpdateFrame(bytes).stream(stream));
+
+                windowUpdateFrameWritten = true;
```

Comment:
`io.netty.handler.codec.http2.Http2FrameCodec#consumeBytes` may return `false`, but `windowUpdateFrameWritten` is always true

Reply:
There might be some more improvements we can make but I would say we can do another PR for this.